recipe-server/contract-tests/test_selfrepair.py

```diff
@@ -0,0 +1,28 @@
+def parse_csp(header):
+    directives = {}
+    parts = [p.strip() for p in header.split(';')]
+    for part in parts:
+        if ' ' in part:
+            directive, data = part.split(' ', 1)
+        else:
+            directive = part
+            data = True
+        directives[directive] = data
+    return directives
+
+
+def test_selfrepair_csp_allows_framing(conf, requests_session):
+    r = requests_session.get(conf.getoption('server') + '/en-US/repair')
+    r.raise_for_status()
+    if 'Content-Security-Policy' not in r.headers:
+        # no CSP, no problem
```

Comment:
I agree that not having CSP should be a test failure. However, this test is very narrowly scoped, only dealing with the bug I'm trying to fix here, which is critical functionality not loading due to an overly aggressive CSP policy.